x/auth/ante.go

```diff
@@ -164,7 +179,21 @@ func processSig(
 		return nil, sdk.ErrInternal("setting PubKey on signer's account").Result()
 	}
 
-	consumeSignatureVerificationGas(ctx.GasMeter(), sig.Signature, pubKey, params)
+	if simulate {
+		// Simulated txs should not contain a signature and are not required to
+		// contain a pubkey, so we must account for tx size of including a
+		// StdSignature (Amino encoding) and simulate gas consumption
+		// (assuming a SECP256k1 simulation key).
+		simSig := StdSignature{PubKey: pubKey}
```

Comment:
In the multisig case, gas estimate would be wrong here

Reply:
we don't support multi-sig for estimation afaik -- correct me if I'm wrong.

Reply:
Can we with a flag or something?

Reply:
@cwgoes multisig is tricky nonetheless - signature verification strictly depends on the signature itself, i.e. given a N-of-M multikey and a multisig signature with P sub sigs, all sub signatures get always processed/verified, even when N < P. Knowing the original N-of-M key might help obtaining a more accurate estimate (i.e. you know that you would process at least N signature), though it does not solve the problem.

Reply:
How about just estimating the upper bound by default?

(I agree that the problem is tricky but we should optimize for the average use case and relatively low cost of paying a bit extra gas)

Reply:
Updated to account for multisigs.